# 1. Importando bibliotecas para utilização no projeto

In [1]:
import pandas as pd
import numpy as np

# 2. Importando e analisando os dados coletados

Dados sobre os casos de câncer no Brasil (2005 - 2019) disponíveis no site do INCA (Instituto Nacional de Câncer): <https://www.inca.gov.br/>.

In [28]:
dataset_1 = pd.read_csv('INCA_2005_2011.csv', sep = ';', encoding = 'latin-1', low_memory = False)
dataset_2 = pd.read_csv('INCA_2012_2019.csv', sep = ';', encoding = 'latin-1', low_memory = False)

In [29]:
dataset_1.head(7)

Código do Paciente           Nome do RCBP       Sexo Data de Nascimento  \
0            672054.0  RCBP CAMPINAS-UNICAMP  MASCULINO         16/03/1975   
1            673745.0  RCBP CAMPINAS-UNICAMP  MASCULINO         03/07/1934   
2            676350.0  RCBP CAMPINAS-UNICAMP  MASCULINO         18/11/1933   
3            679877.0  RCBP CAMPINAS-UNICAMP   FEMININO         30/11/1947   
4            676553.0  RCBP CAMPINAS-UNICAMP  MASCULINO         30/01/1953   
5            679345.0  RCBP CAMPINAS-UNICAMP  MASCULINO         09/10/1930   
6            682806.0  RCBP CAMPINAS-UNICAMP  MASCULINO         02/05/1946   

   Idade Raca/Cor Nacionalidade Naturalidade Estado Naturalidade  \
0   30.0   BRANCO        BRASIL           SÃO PAULO          NaN   
1   71.0   BRANCO        BRASIL           SÃO PAULO          NaN   
2   71.0   BRANCO        BRASIL           SÃO PAULO          NaN   
3   57.0    PARDA        BRASIL             ALAGOAS          NaN   
4   52.0   BRANCO        BRASIL           SÃO PAULO          NaN   
5   74.0   BRANCO        BRASIL       Não informado          NaN   
6   59.0   BRANCO        BRASIL      SANTA CATARINA          NaN   

   Grau de Instrução  ...    Lateralidade         Estadiamento  TNM  \
0     SEM INFORMAÇÃO  ...  SEM INFORMAÇÃO  99 - SEM INFORMAÇÃO  999   
1     SEM INFORMAÇÃO  ...  SEM INFORMAÇÃO  99 - SEM INFORMAÇÃO  999   
2  SUPERIOR COMPLETO  ...  SEM INFORMAÇÃO  99 - SEM INFORMAÇÃO  999   
3     SEM INFORMAÇÃO  ...  SEM INFORMAÇÃO  99 - SEM INFORMAÇÃO  999   
4     SEM INFORMAÇÃO  ...  SEM INFORMAÇÃO  99 - SEM INFORMAÇÃO  999   
5     SEM INFORMAÇÃO  ...  SEM INFORMAÇÃO  99 - SEM INFORMAÇÃO  999   
6     SEM INFORMAÇÃO  ...  SEM INFORMAÇÃO  99 - SEM INFORMAÇÃO  999   

  Status Vital Tipo do Obito Data do Óbito Data de Último Contato  \
0        MORTO        CÂNCER    07/10/2005             07/10/2005   
1        MORTO        CÂNCER    21/10/2005             21/10/2005   
2        MORTO        CÂNCER    06/11/2005             06/11/2005   
3        MORTO        CÂNCER    15/05/2005             15/05/2005   
4        MORTO        CÂNCER    24/02/2005             24/02/2005   
5        MORTO        CÂNCER    01/10/2005             01/10/2005   
6        MORTO        CÂNCER    27/08/2005             27/08/2005   

  Data de Diagnostico  Metástase à distância Unnamed: 37  
0          07/10/2005                    NaN         NaN  
1          21/10/2005                    NaN         NaN  
2          06/11/2005                    NaN         NaN  
3          15/05/2005                    NaN         NaN  
4          24/02/2005                    NaN         NaN  
5          01/10/2005                    NaN         NaN  
6          27/08/2005                    NaN         NaN  

[7 rows x 38 columns]

In [30]:
uniao_dados = pd.merge(dataset_1,dataset_2, how = 'outer')

# 3. Selecionando os dados que serão armazenados em um novo arquivo .csv

Como o arquivo original excede o tamanho suportado pelo GitHub e também possui diversas informação que não serão usadas no projeto, então há a necessidade de escolher os dados que farão parte de um novo dataframe e de um novo .csv mais compacto e objetivo.

In [31]:
uniao_dados.columns

Index(['Código do Paciente', 'Nome do RCBP', 'Sexo', 'Data de Nascimento',
       'Idade', 'Raca/Cor', 'Nacionalidade', 'Naturalidade Estado',
       'Naturalidade', 'Grau de Instrução', 'Estado Civil', 'Código Profissão',
       'Nome Profissão', 'Estado Endereço', 'Cidade Endereço',
       'Descrição da Topografia', 'Código da Topografia',
       'Descrição da Morfologia', 'Código da Morfologia',
       'Descrição da Doenca', 'Código da Doenca',
       'Descrição da Doenca Infantil', 'Código da Doenca Infantil',
       'Descrição da Doenca Adulto Jovem', 'Código da Doenca Adulto Jovem',
       'Indicador de Caso Raro', 'Meio de Diagnostico', 'Extensão',
       'Lateralidade', 'Estadiamento', 'TNM', 'Status Vital', 'Tipo do Obito',
       'Data do Óbito', 'Data de Último Contato', 'Data de Diagnostico',
       'Metástase à distância', 'Unnamed: 37'],
      dtype='object')

In [32]:
uniao_dados2 = uniao_dados[[
    'Código do Paciente',
    'Nome do RCBP',
    'Sexo',
    'Idade',
    'Raca/Cor',
    'Nacionalidade',
    'Naturalidade Estado',
    'Estado Endereço',
    'Data de Diagnostico',
    'Data do Óbito'
]]

In [33]:
uniao_dados2.head()

Código do Paciente           Nome do RCBP       Sexo  Idade Raca/Cor  \
0            672054.0  RCBP CAMPINAS-UNICAMP  MASCULINO   30.0   BRANCO   
1            673745.0  RCBP CAMPINAS-UNICAMP  MASCULINO   71.0   BRANCO   
2            676350.0  RCBP CAMPINAS-UNICAMP  MASCULINO   71.0   BRANCO   
3            679877.0  RCBP CAMPINAS-UNICAMP   FEMININO   57.0    PARDA   
4            676553.0  RCBP CAMPINAS-UNICAMP  MASCULINO   52.0   BRANCO   

  Nacionalidade Naturalidade Estado Estado Endereço Data de Diagnostico  \
0        BRASIL           SÃO PAULO       SÃO PAULO          07/10/2005   
1        BRASIL           SÃO PAULO       SÃO PAULO          21/10/2005   
2        BRASIL           SÃO PAULO       SÃO PAULO          06/11/2005   
3        BRASIL             ALAGOAS       SÃO PAULO          15/05/2005   
4        BRASIL           SÃO PAULO       SÃO PAULO          24/02/2005   

  Data do Óbito  
0    07/10/2005  
1    21/10/2005  
2    06/11/2005  
3    15/05/2005  
4    24/02/2005

## Renomeando as colunas

In [51]:
uniao_dados3 = uniao_dados2.rename(columns = {
    'Código do Paciente': 'codigo_paciente',
    'Nome do RCBP': 'local_atendimento',
    'Sexo':'sexo', 
    'Idade':'idade',
    'Raca/Cor': 'cor',
    'Nacionalidade': 'nacionalidade',
    'Naturalidade Estado': 'estado_nascimento',
    'Estado Endereço': 'estado',
    'Data de Diagnostico': 'data',
    'Data do Óbito': 'data_obito'
})

uniao_dados3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1305823 entries, 0 to 1305822
Data columns (total 10 columns):
 #   Column             Non-Null Count    Dtype  
---  ------             --------------    -----  
 0   codigo_paciente    1253141 non-null  float64
 1   local_atendimento  1305823 non-null  object 
 2   sexo               1305823 non-null  object 
 3   idade              1242934 non-null  float64
 4   cor                1305823 non-null  object 
 5   nacionalidade      764573 non-null   object 
 6   estado_nascimento  767039 non-null   object 
 7   estado             1263373 non-null  object 
 8   data               1263728 non-null  object 
 9   data_obito         417396 non-null   object 
dtypes: float64(2), object(8)
memory usage: 109.6+ MB


In [64]:
uniao_dados3.shape

(1305823, 10)

## Analisando se há pacientes duplicados

In [52]:
pacientes_duplicados = uniao_dados3.codigo_paciente.value_counts()

In [55]:
pacientes_duplicados

525076.0     28
773789.0     27
490532.0     25
885370.0     23
358376.0     22
             ..
1050592.0     1
1050587.0     1
1050589.0     1
1046945.0     1
1398299.0     1
Name: codigo_paciente, Length: 957958, dtype: int64

### Observação dos dados duplicados

Percebe-se que as duplicações estão na mesma localidade. Portanto, trata-se realmente do mesmo paciente e poderá ser excluído para análise.

In [56]:
uniao_dados3.query("codigo_paciente == 358376.0")

codigo_paciente local_atendimento       sexo  idade     cor  \
332091         358376.0      RCBP ARACAJU  MASCULINO   81.0  BRANCO   
334324         358376.0      RCBP ARACAJU  MASCULINO   82.0  BRANCO   
336509         358376.0      RCBP ARACAJU  MASCULINO   83.0  BRANCO   
336863         358376.0      RCBP ARACAJU  MASCULINO   83.0  BRANCO   
336864         358376.0      RCBP ARACAJU  MASCULINO   83.0  BRANCO   
339262         358376.0      RCBP ARACAJU  MASCULINO   84.0  BRANCO   
342076         358376.0      RCBP ARACAJU  MASCULINO   86.0  BRANCO   
344445         358376.0      RCBP ARACAJU  MASCULINO   86.0  BRANCO   
344446         358376.0      RCBP ARACAJU  MASCULINO   86.0  BRANCO   
344447         358376.0      RCBP ARACAJU  MASCULINO   86.0  BRANCO   
344448         358376.0      RCBP ARACAJU  MASCULINO   86.0  BRANCO   
344449         358376.0      RCBP ARACAJU  MASCULINO   86.0  BRANCO   
344450         358376.0      RCBP ARACAJU  MASCULINO   86.0  BRANCO   
344886         358376.0      RCBP ARACAJU  MASCULINO   87.0  BRANCO   
347151         358376.0      RCBP ARACAJU  MASCULINO   87.0  BRANCO   
982028         358376.0      RCBP ARACAJU  MASCULINO   88.0  BRANCO   
983286         358376.0      RCBP ARACAJU  MASCULINO   88.0  BRANCO   
983807         358376.0      RCBP ARACAJU  MASCULINO   88.0  BRANCO   
983808         358376.0      RCBP ARACAJU  MASCULINO   88.0  BRANCO   
986913         358376.0      RCBP ARACAJU  MASCULINO   89.0  BRANCO   
992840         358376.0      RCBP ARACAJU  MASCULINO   91.0  BRANCO   
992841         358376.0      RCBP ARACAJU  MASCULINO   91.0  BRANCO   

        nacionalidade estado_nascimento   estado        data data_obito  
332091  Não informado     Não informado  SERGIPE  21/07/2005        NaN  
334324  Não informado     Não informado  SERGIPE  23/02/2006        NaN  
336509  Não informado     Não informado  SERGIPE  20/07/2007        NaN  
336863  Não informado     Não informado  SERGIPE  20/07/2007        NaN  
336864  Não informado     Não informado  SERGIPE  17/08/2007        NaN  
339262  Não informado     Não informado  SERGIPE  07/02/2008        NaN  
342076  Não informado     Não informado  SERGIPE  23/12/2010        NaN  
344445  Não informado     Não informado  SERGIPE  26/07/2010        NaN  
344446  Não informado     Não informado  SERGIPE  26/07/2010        NaN  
344447  Não informado     Não informado  SERGIPE  26/07/2010        NaN  
344448  Não informado     Não informado  SERGIPE  28/10/2010        NaN  
344449  Não informado     Não informado  SERGIPE  28/10/2010        NaN  
344450  Não informado     Não informado  SERGIPE  23/12/2010        NaN  
344886  Não informado     Não informado  SERGIPE  17/05/2011        NaN  
347151  Não informado     Não informado  SERGIPE  17/05/2011        NaN  
982028  Não informado     Não informado  SERGIPE  06/06/2012        NaN  
983286  Não informado     Não informado  SERGIPE  06/03/2012        NaN  
983807  Não informado     Não informado  SERGIPE  06/03/2012        NaN  
983808  Não informado     Não informado  SERGIPE  06/06/2012        NaN  
986913  Não informado     Não informado  SERGIPE  03/12/2013        NaN  
992840  Não informado     Não informado  SERGIPE  11/02/2015        NaN  
992841  Não informado     Não informado  SERGIPE  11/02/2015        NaN

### Preparando os dados da coluna 'codigo_paciente' para poder excluir os duplicados


In [57]:
uniao_dados3.isnull().sum()

codigo_paciente       52682
local_atendimento         0
sexo                      0
idade                 62889
cor                       0
nacionalidade        541250
estado_nascimento    538784
estado                42450
data                  42095
data_obito           888427
dtype: int64

Há **52682** elementos nulos na coluna do código do paciente.

**Apenas os dados duplicados dos códigos do paciente que serão deletados. Os outros serão analisados e trabalhados posteriormente.**

In [58]:
uniao_dados3['codigo_paciente'] = uniao_dados3['codigo_paciente'].fillna(value = 0)

Os dados do código está em 'float' e os valores adicionados são 'int', por isso há a necessidade de modificar toda a coluna para poder excluir as duplicatas.

In [59]:
uniao_dados3['codigo_paciente'] = uniao_dados3['codigo_paciente'].apply(int)

In [60]:
uniao_dados4 = uniao_dados3.drop_duplicates(subset='codigo_paciente', keep='first')

In [61]:
uniao_dados4.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 957959 entries, 0 to 1305822
Data columns (total 10 columns):
 #   Column             Non-Null Count   Dtype  
---  ------             --------------   -----  
 0   codigo_paciente    957959 non-null  int64  
 1   local_atendimento  957959 non-null  object 
 2   sexo               957959 non-null  object 
 3   idade              906892 non-null  float64
 4   cor                957959 non-null  object 
 5   nacionalidade      570600 non-null  object 
 6   estado_nascimento  572603 non-null  object 
 7   estado             957923 non-null  object 
 8   data               957958 non-null  object 
 9   data_obito         328005 non-null  object 
dtypes: float64(1), int64(1), object(8)
memory usage: 80.4+ MB


In [63]:
uniao_dados4.shape

(957959, 10)

Anteriormento, o dataframe tinha 1305823 linhas e tem **957959** agora.

# 4. Exportando os dados para um arquivo .csv

In [66]:
uniao_dados4.to_csv("data_cancer_2005_2019.csv")